src/rbtree.rs

```diff
@@ -1927,10 +2063,56 @@ where
             } else {
                 self.insert_root(new);
             }
-            CursorMut {
-                current: Some(new),
-                tree: self,
-            }
+
+            new
+        }
+    }
+
+    /// Inserts a new element into the `RBTree`.
+    ///
+    /// The new element will be inserted at the correct position in the tree
+    /// based on its key.
+    ///
+    /// Returns a mutable cursor pointing to the newly added element.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the new element is already linked to a different intrusive
+    /// collection.
+    #[inline]
+    pub fn insert<'a>(&'a mut self, val: <A::PointerOps as PointerOps>::Pointer) -> CursorMut<'_, A>
+    where
+        <A as KeyAdapter<'a>>::Key: Ord,
+    {
+        CursorMut {
+            current: Some(self.insert_internal(val)),
+            tree: self,
+        }
+    }
+
+    /// Inserts a new element into the `RBTree`.
+    ///
+    /// The new element will be inserted at the correct position in the tree
+    /// based on its key.
+    ///
+    /// Consumes the `RBTree` and returns an owning cursor pointing to the newly added element.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the new element is already linked to a different intrusive
+    /// collection.
+    #[inline]
+    pub fn insert_owning<'a>(
```

Comment:
I don't think this method is sufficiently useful to justify its inclusion. I would prefer not having it unless you have a specific use case for it.

Reply:
Well given that a `CursorMut` cannot be made into a `CursorOwning` it's pretty convenient, but not strictly necessary I guess?